File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<Integer>();

        for (int n : nums) {
            set.add(n);
        }

        int longest = 0;

        for (int i = 0; i < nums.length; i++) {
            if (!set.contains(nums[i] - 1)) {
                int length = 1;
                while (set.contains(nums[i] + length)) {
                    length++;
                }
                longest = Math.max(longest, length);
            }
        }

        return longest;
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-2.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]

        memo = {}

        def dfs(i, flag):
            if i >= len(nums):
                return 0

            if flag and i == len(nums) - 1:
                return 0

            if (i, flag) in memo:
                return memo[(i, flag)]

            memo[(i, flag)] = max(nums[i] + dfs(i + 2, flag), dfs(i + 1, flag))
            return memo[(i, flag)]
        
        return max(dfs(0, True), dfs(1, False))